charts/chart_deps/app/common/templates/deployment.yaml

```diff
@@ -23,7 +23,7 @@ spec:
     metadata:
       {{- with .Values.podAnnotations }}
       annotations:
-        {{- toYaml . | nindent 8 }}
+        {{- tpl (toYaml .) $ | nindent 8 }}
```

Comment:
what advantages does this provide?

Reply:
yeah so the goal is to move env/company related stuff out off apps. With tpl function we can do something like
```
annotations:
      iam.gke.io/gcp-service-account: overlai-file-worker@{{ .Values.global.company.gcp.project_id }}.iam.gserviceaccount.com
```

Reply:
on the other hand seems you are right because the example i used is related to ServiceAccount obj not deployment. I dont see any reason to use tpl  for deployments atm